api/channel.go

```diff
@@ -1293,3 +1214,66 @@ func autocompleteChannels(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write([]byte(channels.ToJson()))
 }
+
+func viewChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	view := model.ChannelViewFromJson(r.Body)
+
+	go func() {
+		if err := SetActiveChannel(c.Session.UserId, view.ChannelId); err != nil {
+			l4g.Error(err.Error())
+		}
+	}()
+
+	if len(view.ChannelId) > 0 {
+
+		if view.Time == 0 {
+			if result := <-Srv.Store.Channel().UpdateLastViewedAt(view.ChannelId, c.Session.UserId); result.Err != nil {
+				c.Err = result.Err
+				return
+			}
+		} else {
+			if result := <-Srv.Store.Channel().SetLastViewedAt(view.ChannelId, c.Session.UserId, view.Time); result.Err != nil {
```

Comment:
We were already trusting the user time in the original `set_last_viewed_at` API, I just moved the functionality here. What's wrong with trusting the user time? It's only setting their own unreads right